app/uk/gov/voa/propertylinking/errorhandler/CustomErrorHandler.scala

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.voa.propertylinking.errorhandler
+
+import java.time.{Instant, LocalDateTime, ZoneId}
+
+import config.ApplicationConfig
+import javax.inject.Inject
+import play.api.Logger
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.mvc.Request
+import play.twirl.api.Html
+import uk.gov.hmrc.http.HeaderNames
+import uk.gov.hmrc.play.bootstrap.http.FrontendErrorHandler
+
+class CustomErrorHandler @Inject()(implicit val messagesApi: MessagesApi, appConfig: ApplicationConfig) extends FrontendErrorHandler with I18nSupport {
+
+  val logger: Logger = Logger(this.getClass)
+
+  override def standardErrorTemplate(pageTitle: String, heading: String, message: String)(implicit request: Request[_]): Html =
+    views.html.errors.error(pageTitle, heading, message)
+
+  override def internalServerErrorTemplate(implicit request: Request[_]): Html = {
+    views.html.errors.technicalDifficulties(extractErrorReference(request), getDateTime)
+  }
+
+  private def getDateTime: LocalDateTime = {
+    val instant = Instant.ofEpochMilli(System.currentTimeMillis)
+    LocalDateTime.ofInstant(instant, ZoneId.of("Europe/London"))
+  }
+
+  private def extractErrorReference(request: Request[_]): Option[String] = {
+    request.headers.get(HeaderNames.xRequestId) map {
+      _.split("-")(2)
```

Comment:
where did this come from? is copy-paste from some other service? doesn't seem safe to do this at all...  you don't know if there are 2 hyphens in that header value. `split` you could wrap the split...(2) in a Try and then .toOption it in a `flatMap`

```scala
request.headers.get(HeaderNames.xRequestId).flatMap ( v => Try(v.split("-")(2)).toOption)
```

Reply:
request-id is wanted to be displayed on the screen this is really BROKEN but I copied it from the other place it sat we need to discuss the best way to get the request-id out of the headers for all frontend services.